### PR TITLE
Phase 4 PoC

### PR DESCRIPTION
**Pull request by vynpt**
Our Proof of Concept is a dashboard with associated code. The "app.py" inside the demoapp folder is the Python script that includes codes of LLM model and Streamlit dashboard. Please run "app.py", we also have a document "DemoApp_Guideline.md" to provide instruction to run this file.

### CODE REVIEW

**Review by gbhandarwar — COMMENTED** [review]
Hi, guys I just found more prompts in a pdf file, should i also add it here again? Or can i just include it in the research docs later on so it does not eat up space here.

**Comment by vynpt** [comment]
@gbhandarwar  yes just add them to the "prompts.md" inside Prompt Engineering folder.

**Review by trgardos — APPROVED** [review]

File: demoapp/DemoApp_Guideline.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+In the file "app.py" in "demoapp" folder:
+* ```pip install streamlit```
+* Install all imported libraries
+* Using Anaconda or create an environment to run streamlit  
+    To create env:  
+    ```python3 -m venv env```   
+    ```source ./env/bin/activate```
+* Inside function "generate_response(text, title)", insert your own OpenAI key.
+
```
Comment (oindrillac) [inline demoapp/DemoApp_Guideline.md]: Instruction to the app streamlit run app.py missing

File: demoapp/DemoApp_Guideline.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Guideline to run the DemoApp using Streamlit
+
+In the file "app.py" in "demoapp" folder:
+* ```pip install streamlit```
+* Install all imported libraries
+* Using Anaconda or create an environment to run streamlit  
+    To create env:  
+    ```python3 -m venv env```   
+    ```source ./env/bin/activate```
+* Inside function "generate_response(text, title)", insert your own OpenAI key.
```
Comment (oindrillac) [inline demoapp/DemoApp_Guideline.md]: Editing the file to insert the api key offers a complicated user interface, can we accept the API key as an input in the streamlit app?

**Comment by oindrillac** [comment]
Screenshot 2023-11-14 at 11 29 52 AM
Getting this error with the third bill on the streamlit app

File: demoapp/DemoApp_Guideline.md (hunk 1 above)
Comment (trgardos) [reply demoapp/DemoApp_Guideline.md]: Should say that demoapp/requirements.txt should be installed, but I get an error when installing.

File: demoapp/DemoApp_Guideline.md (hunk 2 above)
Comment (vynpt) [reply demoapp/DemoApp_Guideline.md]: we did this

File: demoapp/DemoApp_Guideline.md (hunk 1 above)
Comment (vynpt) [reply demoapp/DemoApp_Guideline.md]: we updated the requirement.txt and the Demo_Guideline.md fild

**Pull request merged**